src/main/java/org/opentripplanner/model/plan/pagecursor/PageCursorFactory.java

```diff
@@ -48,27 +48,18 @@ public PageCursorFactory withOriginalSearch(
   }
 
   /**
-   * Set the start and end time for removed itineraries. The current implementation uses the FIRST
-   * removed itinerary, but this can in some cases lead to missed itineraries in the next search.
-   * So, we will document here what should be done.
-   * <p>
-   * For case {@code depart-after-crop-sw} and {@code arrive-by-crop-sw-reversed-filter} the {@code
-   * startTime} should be the EARLIEST departure time for all removed itineraries.
-   * <p>
-   * For case {@code depart-after-crop-sw-reversed-filter} and {@code arrive-by-crop-sw} the {@code
-   * startTime} should be the LATEST departure time for all removed itineraries.
-   * <p>
-   * The {@code endTime} should be replaced by removing duplicates between the to pages. This can
-   * for example be done by including a hash for each potential itinerary in the token, and make a
-   * filter to remove those in the following page response.
+   * If there were itineraries removed in the current search because the numItineraries parameter
+   * was used, then we want to allow the caller to move within some of the itineraries that were
+   * removed in the next and previous pages. This means we will use information from when we cropped
+   * the list of itineraries to create the new search encoded in the page cursors.
    *
-   * @param startTime is rounded down to the closest minute.
-   * @param endTime   is round up to the closest minute.
+   * @param numItinerariesFilterResults the result from the {@code NumItinerariesFilter}
    */
-  public PageCursorFactory withRemovedItineraries(Instant startTime, Instant endTime) {
+  public PageCursorFactory withRemovedItineraries(
+    NumItinerariesFilterResults numItinerariesFilterResults
```

Comment:
This is creating a dependency from the `model` to the `filterchain`, but the dependency should be the other way around. If you think the method have to many arguments you can make a request object(`PageCursorParameters`).  The `PageCursorParameters` can be an interface or POJO/record. The filter can return the `PageCursorParameters`(option A), since this is the only purpose of the retuned type. Det downside is that the filter now is responsible for mapping into the `PageCursorParameters` - it is ok(pragmatic) if the mapping is simple. The parameter type can not include `ListSection`, so you will need to find a solution that as well. A slightly better option (B) is to let the `NumItinerariesFilterResults` inherit the `PageCursorParameters` _interface_. This should tell the reader of the code about the relationship and usage. It also make it possible to customize the `NumItinerariesFilterResults` for other needs within the filter-chan(I am not sure if that is needed). The last option(C) is to make a mapper somewhere between. This is the most verbose solution, but also the cleanest, because every pice has a singel purpose (singel responsibility) and become simpler to test.

Reply:
I'll opt for B. Information about the removed itineraries is used in a couple of other places: setting the new adjusted search window and generating the search metadata, so it will be useful to have them around.